bindings/r/CMakeLists.txt

```diff
@@ -39,5 +41,4 @@ ROOT_ADD_CXX_FLAG(_R_FLAGS -Wno-overloaded-virtual)
 separate_arguments(_R_FLAGS)
 
 target_compile_options(RInterface PUBLIC ${_R_FLAGS})
-target_include_directories(RInterface BEFORE PUBLIC ${R_INCLUDE_DIRS})
```

Comment:
I uses `include_directories` when both library and dictionary requires such includes.
Otherwise one has to use `cmake generator expression` which has limited functionality
Please see my comments in mattermost

Reply:
And of course, every target defined after `include_directories` get automatically assigned these includes to target properties.

Reply:
I cannot find your mattermost comments. Are you saying that ROOT_GENERATE_DICTIONARY fails to 'inherit' the `target_include_directories`?

Reply:
I already answered. ROOT_GENERATE_DICTIONARY called before `target_include_directories` and therefore `target_property` does not return that expected

Reply:
> ROOT_GENERATE_DICTIONARY called before `target_include_directories`

Let's fix that.

Reply:
Then you have to split ROOT_STANDARD_LIBRARY macro - on library build and dictionary build. Or my original solution - extra `INCLIDES` parameter, which set `target_include_directories` before calling `ROOT_GENERATE_DICTIONARY`